### ISSUE TITLE
feat: Insurance domain LoRA fine-tuning for qwen3:8b

### ISSUE THREAD

**Issue by jjob-spec** [PaultheAICoder/VE-RAG-System#327]
## Reference
Assessment: `docs/VE-RAG-System-Assessment-021826.md` — Phase 3, Item #11

## Problem

**Current State:** Generic qwen3:8b has no insurance-specific training. It doesn't understand that:
- "Occurrence Limit" and "General Aggregate" are sub-limits within GL coverage
- "No Coverage" and "Not Offered" have different meanings (excluded vs not available)
- "CGL" = Commercial General Liability, "BOP" = Business Owner's Policy
- "AI endorsement" = Additional Insured endorsement (not artificial intelligence)
- An HOA D&O policy covers board members differently than a corporate D&O

**Required:** A LoRA adapter fine-tuned on insurance Q&A pairs, deployed alongside the base model in Ollama.

### 1. Coverage Summary Q&A Pairs (~200 pairs)
Generated from extracted coverage data:
```json
{"question": "What is the General Liability occurrence limit for Bethany Terrace?",
 "answer": "The GL occurrence limit is $1,000,000 per the LIO 2025-2026 coverage summary.",
 "context": "[coverage summary table text]"}

{"question": "Does Bethany Terrace have earthquake coverage?",
 "answer": "No, earthquake coverage is Not Offered across all quoted carriers (LIO and Philadelphia).",
 "context": "[coverage summary table text]"}
```

### 2. Policy Interpretation Q&A (~150 pairs)
From D&O and GL policy documents:
```json
{"question": "What exclusions apply to the D&O policy?",
 "answer": "The D&O policy excludes: (1) bodily injury/property damage claims, (2) prior or pending litigation, (3) criminal/fraudulent acts, (4) ERISA violations...",
 "context": "[policy exclusion section text]"}
```

### 3. Insurance Terminology (~100 pairs)
Domain vocabulary training:
```json
{"question": "What does 'Defense Cost Outside Limits' mean?",
 "answer": "Defense Cost Outside Limits means the insurer pays legal defense costs in addition to the policy limit, not reducing the available coverage limit. This is favorable compared to 'Defense Cost Inside Limits' where defense costs erode the policy limit."}
```

### 4. Gap Analysis Q&A (~50 pairs)
Coverage gap identification:
```json
{"question": "What coverages should an HOA have that Bethany Terrace is missing?",
 "answer": "Based on the coverage summary, Bethany Terrace lacks: (1) Earthquake coverage (Not Offered), (2) Pollution/Environmental Liability (No Coverage), (3) Building Ordinance coverage (No Coverage), (4) Umbrella/Excess liability ($5M recommended for HOAs). These are standard HOA coverages that should be discussed with the agent."}
```

### 1. Data collection and labeling
- Extract Q&A pairs from test documents using structured extraction
- Generate synthetic Q&A using a stronger model (deepseek-r1:32b) with human review
- Target: 500+ high-quality Q&A pairs across all categories

### 2. LoRA training
```bash
# Using Unsloth or LLaMA-Factory on DGX Spark
python train_lora.py \
    --base_model qwen3:8b \
    --dataset insurance_qa.jsonl \
    --lora_rank 16 \
    --lora_alpha 32 \
    --epochs 3 \
    --learning_rate 2e-4 \
    --output insurance_lora_v1
```

### 3. Deploy adapter in Ollama
```Modelfile
FROM qwen3:8b
ADAPTER ./insurance_lora_v1
PARAMETER temperature 0.1
```

```bash
ollama create qwen3-insurance -f Modelfile
```

### 4. A/B validation
Run all 10 test scenarios with both models:
- qwen3:8b (baseline)
- qwen3-insurance (fine-tuned)
Compare accuracy, domain terminology usage, and confidence scores.

## Expected Improvements
- Better understanding of coverage structure relationships
- Correct interpretation of endorsement schedules
- More accurate gap analysis
- Better handling of insurance abbreviations (CGL, BOP, WC, AI)
- Understanding "No Coverage" vs "Not Offered" vs "Declined" semantics

## Files to Modify
- New: `scripts/training/prepare_insurance_qa.py` — data preparation
- New: `scripts/training/train_lora.py` — training script
- New: `data/training/insurance_qa.jsonl` — training data
- New: `Modelfile.insurance` — Ollama model definition
- `ai_ready_rag/config.py` — add `insurance_model` config option

## Acceptance Criteria
- [ ] 500+ Q&A training pairs covering all 4 categories
- [ ] LoRA adapter trained and deployable via Ollama
- [ ] A/B comparison shows improvement on test scenarios
- [ ] No regression on general knowledge queries
- [ ] Adapter size <100MB (fits on DGX Spark alongside base model)
- [ ] Training reproducible from scripts

## Effort Estimate
**4-6 weeks** (COMPLEX)
- Data collection: 2 weeks
- Training + iteration: 1-2 weeks
- Validation + deployment: 1 week

## Dependencies
- #326 (Coverage Summary extractor) — provides structured training data
- DGX Spark GPU availability for training

**Comment by jjob-spec** [comment]
**Closing — Superseded by VaultIQ architecture.**

The VaultIQ spec (`specs/INSURANCE_AI_PLATFORM_v1.md` v1.1) uses Claude API (Sonnet) for document enrichment and query-time answering (Standard tier). This eliminates the need for LoRA fine-tuning of local models in the near term. Enterprise tier benefits from Claude-enriched chunks (summaries + entities) that significantly improve local model performance without fine-tuning.

LoRA fine-tuning may be revisited as a future optimization for Enterprise tier, but is not on the current roadmap.
